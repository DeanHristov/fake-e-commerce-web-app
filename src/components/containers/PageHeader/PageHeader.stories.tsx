import type { Meta, StoryObj } from '@storybook/react';
import PageHeader, { IPageHeaderProps } from './index';

const meta: Meta<IPageHeaderProps> = {
  title: 'Containers/PageHeader',
  component: PageHeader,
  parameters: {
    layout: 'centered',
  },
  argTypes: {},
};

export default meta;

type Story = StoryObj<IPageHeaderProps>;

export const Primary: Story = {
  render: () => <PageHeader />,
};
