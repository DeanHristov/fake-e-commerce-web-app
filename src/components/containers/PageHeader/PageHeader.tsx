'use client';

import { FC } from 'react';
import Link from 'next/link';
import Button from '../../ui/Button';
import {
  MagnifyingGlassIcon,
  ShoppingCartIcon,
  UserIcon,
} from '@heroicons/react/24/solid';
import Badge from '../../ui/Badge';
import SearchField from '../../ui/SearchField';
import { useRouter } from 'next/navigation';

export interface IPageHeaderProps {}

const PageHeader: FC<IPageHeaderProps> = ({}) => {
  const router = useRouter();
  return (
    <div className="shadow-md px-4">
      <header className="flex flex-col md:flex-row  min-w-[320px] max-w-screen-xl m-auto">
        <div className="w-full py-2 md:w-4/12 text-center md:text-left">
          <Link href={'/'} className="w-full text-black font-bold text-2xl">
            E-commerce
          </Link>
        </div>
        <div className="w-full py-2 self-center md:p-0 md:w-96">
          <SearchField
            placeholder={'Search me...'}
            onChange={(value) => console.log(value)}
            rightIcon={
              <MagnifyingGlassIcon className="w-7 h-7 text-zinc-100" />
            }
          />
        </div>
        <div className="w-full py-2 md:w-4/12 flex justify-center md:justify-end items-center">
          <div className="w-24 md:w-14 border text-center">
            <Link href={'/shopping-card'}>
              <Badge counter={6}>
                <ShoppingCartIcon className="w-7 h-7 text-gray-500" />
              </Badge>
            </Link>
          </div>

          <div className="w-24 md:w-14 text-center">
            <Link href={'/admin'}>
              <Badge>
                <UserIcon className="w-7 h-7 text-gray-500" />
              </Badge>
            </Link>
          </div>

          <div className="w-24 text-center">
            <Button onClick={() => router.push('/login')} title={'Login'} />
          </div>
        </div>
      </header>
    </div>
  );
};

export default PageHeader;
