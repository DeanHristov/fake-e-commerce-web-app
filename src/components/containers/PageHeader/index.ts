import PageHeader, { IPageHeaderProps } from './PageHeader';

export type { IPageHeaderProps };

export default PageHeader;
