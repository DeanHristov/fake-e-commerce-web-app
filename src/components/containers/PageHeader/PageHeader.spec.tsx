import { render } from '@testing-library/react';
import PageHeader from './index';

describe('Containers/PageHeader <PageHeader {...}>', () => {
  it('Should render default state', () => {
    const { container } = render(<PageHeader />);

    expect(container).toMatchSnapshot();
  });
});
