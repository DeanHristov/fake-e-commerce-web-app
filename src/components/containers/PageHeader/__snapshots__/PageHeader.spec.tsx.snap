// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers/PageHeader <PageHeader {...}> Should render default state 1`] = `
<div>
  <div
    class="shadow-md px-4"
  >
    <header
      class="flex flex-col md:flex-row  min-w-[320px] max-w-screen-xl m-auto"
    >
      <div
        class="w-full py-2 md:w-4/12 text-center md:text-left"
      >
        <a
          class="w-full text-black font-bold text-2xl"
          href="/"
        >
          E-commerce
        </a>
      </div>
      <div
        class="w-full h-14 py-2 self-center md:w-96"
      >
        <div
          class="relative w-full rounded-lg h-full border border-gray-500"
        >
          <input
            class="h-full px-6 w-full rounded-lg bg-white text-slate-700 ease-in duration-200 focus:bg-white"
            placeholder="Search me..."
            type="text"
            value=""
          />
          <div
            class="absolute top-1 left-auto right-2"
          >
            <svg
              aria-hidden="true"
              class="w-7 h-7 text-zinc-100"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M10.5 3.75a6.75 6.75 0 100 13.5 6.75 6.75 0 000-13.5zM2.25 10.5a8.25 8.25 0 1114.59 5.28l4.69 4.69a.75.75 0 11-1.06 1.06l-4.69-4.69A8.25 8.25 0 012.25 10.5z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="w-full py-2 md:w-4/12 flex justify-center md:justify-end items-center"
      >
        <div
          class="w-24 md:w-14 border text-center"
        >
          <a
            href="/shopping-card"
          >
            <span
              class="relative inline-block px-2 cursor-pointer"
            >
              <svg
                aria-hidden="true"
                class="w-7 h-7 text-gray-500"
                fill="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M2.25 2.25a.75.75 0 000 1.5h1.386c.17 0 .318.114.362.278l2.558 9.592a3.752 3.752 0 00-2.806 3.63c0 .414.336.75.75.75h15.75a.75.75 0 000-1.5H5.378A2.25 2.25 0 017.5 15h11.218a.75.75 0 00.674-.421 60.358 60.358 0 002.96-7.228.75.75 0 00-.525-.965A60.864 60.864 0 005.68 4.509l-.232-.867A1.875 1.875 0 003.636 2.25H2.25zM3.75 20.25a1.5 1.5 0 113 0 1.5 1.5 0 01-3 0zM16.5 20.25a1.5 1.5 0 113 0 1.5 1.5 0 01-3 0z"
                />
              </svg>
              <span
                class="absolute top-0 right-2 inline-flex items-center justify-center px-2 py-1 text-xs font-bold leading-none text-red-100 transform translate-x-1/2 -translate-y-1/2 bg-red-600 rounded-full"
              >
                6
              </span>
            </span>
          </a>
        </div>
        <div
          class="w-24 md:w-14 text-center"
        >
          <a
            href="/admin"
          >
            <span
              class="relative inline-block px-2 cursor-pointer"
            >
              <svg
                aria-hidden="true"
                class="w-7 h-7 text-gray-500"
                fill="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M7.5 6a4.5 4.5 0 119 0 4.5 4.5 0 01-9 0zM3.751 20.105a8.25 8.25 0 0116.498 0 .75.75 0 01-.437.695A18.683 18.683 0 0112 22.5c-2.786 0-5.433-.608-7.812-1.7a.75.75 0 01-.437-.695z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </a>
        </div>
        <div
          class="w-24 text-center"
        >
          <div
            class=""
          >
            <button
              class="bg-green-600 w-full px-4 py-2 rounded-sm text-white font-bold uppercase active:shadow-md ease-in duration-200 flex justify-center items-center gap-2"
              type="button"
            >
              Login
            </button>
          </div>
        </div>
      </div>
    </header>
  </div>
</div>
`;
